pkg/nodeagent/controller/operatingsystemconfig/changes.go

```diff
@@ -42,6 +50,14 @@ func extractOSCFromSecret(secret *corev1.Secret) (*extensionsv1alpha1.OperatingS
 		return nil, nil, "", fmt.Errorf("unable to decode OSC from secret data key %s: %w", nodeagentv1alpha1.DataKeyOperatingSystemConfig, err)
 	}
 
+	// Ingest the containerd drop-in into the OSC to prevent side effects when containerd.service is changed by extensions too.
+	ingestContainerdEnvironmentDropIn(osc)
```

Comment:
I don't understand why this is here now - I'm a bit surprised to see this in this `changes.go` file.

Reply:
Btw. I think we could also keep this inside [`ReconcileContainerdConfig`](https://github.com/gardener/gardener/blob/66cf1fb01458877be9fe0bc6922e215b4bc6cbf0/pkg/nodeagent/controller/operatingsystemconfig/containerd.go#L36) and move [the section](https://github.com/gardener/gardener/blob/66cf1fb01458877be9fe0bc6922e215b4bc6cbf0/pkg/nodeagent/controller/operatingsystemconfig/reconciler.go#L103-L106) before the calculation of changes in the reconciliation flow.